node/pkg/lifecycle/startup/startup.go

```diff
@@ -187,20 +187,18 @@ func Run() {
 		configureAndCheckIPAddressSubnets(ctx, cli, node, k8sNode)
 		// Configure the node AS number.
 		configureASNumber(node)
+		// Populate a reference to the node based on orchestrator node identifiers.
+		configureNodeRef(node)
+		// Apply the updated node resource.
+		if _, err := CreateOrUpdate(ctx, cli, node); err != nil {
+			log.WithError(err).Errorf("Unable to set node resource configuration")
+			utils.Terminate()
+		}
 	}
 
-	// Populate a reference to the node based on orchestrator node identifiers.
-	configureNodeRef(node)
```

Comment:
So I think the original PR worked because the IP address itself wasn't used by Calico in CALICO_NETWORKING_BACKEND=none mode, so simply not setting it in that mode was OK.

However, the node reference _is_ used in such a mode - namely, it's used when running Calico against an etcd data store (i.e., not via the k8s API) and using another routing solution (e.g., flannel). So I think this will break a small number of clusters if we move this inside that branch.

I think instead we'll need to conditionally call `CreateOrUpdate` based on whether we _need_ to do so.

e.g., track output from each of the `configureX` methods and then only update the Node if one of them signaled that the node object needs updating. 

WDYT?

Reply:
Thanks @caseydavenport for the quick review. I have updated the change based on your recommendation. Does this LGTY?.